src/transformers/models/bart/modeling_bart.py

```diff
@@ -599,8 +599,11 @@ def __init_subclass__(self):
             is used, optionally only the last `decoder_input_ids` have to be input (see `past_key_values`).
 
             For translation and summarization training, `decoder_input_ids` should be provided. If no
-            `decoder_input_ids` is provided, the model will create this tensor by shifting the `input_ids` to the right
-            for denoising pre-training following the paper.
+            `decoder_input_ids` are provided, the model will automatically create this tensor by shifting:
+
+            - the `labels` to the right for text generation if `labels` are provided,
+            - the `input_ids` to the right for denoising pre-training following the paper if no `labels` are provided.
+
```

Comment:
I'm not sure if this statement is correct. BART is pre-trained with sentence-permutation and text-infilling objective. Where the input text is noised and is passed to the encoder and the decoder is tasked to generate the original document. See this figure from the paper.
![Screenshot 2022-02-21 at 12 16 06 PM](https://user-images.githubusercontent.com/27137566/154945352-8d939035-2d1e-4693-9c6f-47695758bfac.png)

So for this task it's not possible to create `decoder_input_ids` by shifting `input_ids`.

This is only True for the sequence-classification task where we feed the same input to both encoder and decoder.
see this figure from the paper: 
![Screenshot 2022-02-21 at 12 16 24 PM](https://user-images.githubusercontent.com/27137566/154945595-cc313308-2ece-4fb4-9048-e896cb4f91bd.png)

cc @patrickvonplaten